orc8r/cloud/go/services/tenants/client_api.go

```diff
@@ -72,17 +72,8 @@ func GetTenant(ctx context.Context, tenantID int64) (*protos.Tenant, error) {
 	if err != nil {
 		return nil, err
 	}
-
 	tenant, err := oc.GetTenant(ctx, &protos.GetTenantRequest{Id: tenantID})
-	if err != nil {
-		switch {
-		case status.Convert(err).Code() == codes.NotFound:
-			return tenant, merrors.ErrNotFound
-		default:
-			return tenant, err
-		}
-	}
-	return tenant, nil
+	return tenant, errorHandling(err)
```

Comment:
General: we still want to have the "if err != nil" block in this code. It's just the mapping between error types that we want to encapsulate

Also, when there's an err, we want to return zero-y values for all the non-err return vars. That's a good general default, because we don't want callers to become reliant on this implicit/unofficial part of the function contract

So it would be e.g.

```
if err != nil {
	return nil, mapErr(err)
}